### ISSUE TITLE
Hang main form before updating

### ISSUE THREAD

**Issue by radiaku** [riderkick/FMD#51]
Recently after updating to fmd_0.9.16.1 ( fresh download as instructed )

Each time I clicking on update. main form always hang around 30 second - 1minute. after that main form is closed and starting calling form update. 

This never happen in previous version, updating is going smoothly . :)

Sorry for nagging, I just felt dont right :+1:

**Comment by riderkick** [comment]
How about from 0.9.16.3 to 0.9.16.4?

**Comment by radiaku** [comment]
As I remember, its going smoothly when update, closing without delay and starting download UI instantly.

**Comment by riderkick** [comment]
So, there is no problem anymore?

**Comment by radiaku** [comment]
Ehhh.. sorry, what I mean in comment before is in version 0.9.16.1 .. Updater is fine...  

after manually download version fmd_0.9.16.1 and all version after that... fmd main form hang for second before closing and firing UPDATE gui. :)

**Comment by riderkick** [comment]
That's probably because there is still another thread running in the background, I just change from "CloseAllThreadAndWait>Updater>Close" to "Updater>CloseAllThreadAndWait>Close".
Wasn't that supposed to speed up launching the updater?

**Comment by radiaku** [comment]
Well, maybe yes, but what thread?...... What I usually do is, launching fmd.exe -> waiting update message pop out-> clicking update -> Main form closed -> updater gui. previous version worked without delay. 

after fmd_0.9.16.1,  launching fmd.exe -> waiting update message pop out-> clicking update -> Main form HANG ( 60seconds or less) -> Main form closed -> updater gui.

well, I dont have enough knowledge in fpc, so I dont know how to help. FMD is huge design and complex software. Maybe its would speed up the launching :)

**Comment by riderkick** [comment]
if you can compile fmd, please check my latest commit, I added closeallthread without waitfor and increase hearbeatrate to speed up thread terminate respond.

**Comment by riderkick** [comment]
BTW, I don't feel any slow down here. Is there by any chance your antivirus interfere with updater, before launching updater, it will be renamed to old_updater, maybe your antivirus hold and check it first before launching.

**Comment by radiaku** [comment]
Okay I know what happen. its the antivirus is the culprit. A apologize this was my mistake.

antivirus is off
https://www.dropbox.com/s/ivoij3qg14ooqwd/Video_2015-06-13_033036.wmv?dl=0

antivirus is on
https://www.dropbox.com/s/rkvgnbls802gz84/Video_2015-06-13_033308.wmv?dl=0


but what happen in previous version, update is fine.. , maybe its related with renaming old_updater.exe. :) .

**Comment by riderkick** [comment]
Renaming to old_updater have been there since forever. It was postponed prior migrating to github.
the updater renamed to old_updater so the "updater.exe" can be replaced with new updater from update package. old_updater will be deleted when relaunching fmd.
There is many logic that used by antivirus, some of them keep track of the crc/md5 of the files and just check new files. Maybe your updater never get checked before. Or your antivirus setting is to high, paranoid mode?

**Comment by radiaku** [comment]
not at all, my antivirus is on default level, and if have problem on antivirus. why only after version  fmd_0.9.16.1 .  Well i guess i can live on it

**Comment by riderkick** [comment]
Well like what I said above, probably you never run the updater (download list from server). Maybe file check only triggered only when you launch the app.There is indeed some changes to updater since 0.9.16.2, you can see on the changelog or the commits.

**Comment by radiaku** [comment]
well, I have already running the updater ( download list from server, yesterday ). I think its not on launched, maybe on renaming and creating old_updater.exe :)

**Comment by riderkick** [comment]
We have been renaming to old_updater.exe since forever. The problem is with your antivirus logic to recognize new file. Just exclude the file if you don't want it to be scanned.

**Comment by radiaku** [comment]
Well, as I said... I can live on it... :D and already excluded it..... thanks for your effort
